gpt_engineer/applications/cli/main.py

```diff
@@ -0,0 +1,216 @@
+- For azure_endpoint, provide the endpoint for Azure OpenAI service.
+
+"""
+
+import os
+from pathlib import Path
+
+import openai
+import typer
+from dotenv import load_dotenv
+
+from gpt_engineer.core.default.on_disk_repository import OnDiskRepository
+from gpt_engineer.core.ai import AI
+from gpt_engineer.core.default.paths import PREPROMPTS_PATH
+from gpt_engineer.applications.cli.file_selector import ask_for_files, get_all_code
+from gpt_engineer.tools.custom_steps import (
+    lite_gen,
+    clarified_gen,
+    self_heal,
+    vector_improve,
+)
+from gpt_engineer.core.default.git_version_manager import GitVersionManager
+from gpt_engineer.core.default.steps import gen_code, execute_entrypoint, improve
+from gpt_engineer.applications.cli.cli_agent import CliAgent
+from gpt_engineer.applications.cli.collect import collect_and_send_human_review
+from gpt_engineer.core.preprompts_holder import PrepromptsHolder
+import logging
+
+app = typer.Typer()  # creates a CLI app
+
+
+def load_env_if_needed():
+    if os.getenv("OPENAI_API_KEY") is None:
+        load_dotenv()
+    if os.getenv("OPENAI_API_KEY") is None:
+        # if there is no .env file, try to load from the current working directory
+        load_dotenv(dotenv_path=os.path.join(os.getcwd(), ".env"))
+    openai.api_key = os.getenv("OPENAI_API_KEY")
+
+
+def load_prompt(input_repo: OnDiskRepository, improve_mode):
+    if input_repo.get("prompt"):
+        return input_repo.get("prompt")
+
+    if not improve_mode:
+        input_repo["prompt"] = input(
+            "\nWhat application do you want gpt-engineer to generate?\n"
+        )
+    else:
+        input_repo["prompt"] = input("\nHow do you want to improve the application?\n")
+    return input_repo.get("prompt")
+
+
+def get_preprompts_path(use_custom_preprompts: bool, input_path: Path) -> Path:
+    original_preprompts_path = PREPROMPTS_PATH
+    if not use_custom_preprompts:
+        return original_preprompts_path
+
+    custom_preprompts_path = input_path / "preprompts"
+    if not custom_preprompts_path.exists():
+        custom_preprompts_path.mkdir()
+
+    for file in original_preprompts_path.glob("*"):
+        if not (custom_preprompts_path / file.name).exists():
+            (custom_preprompts_path / file.name).write_text(file.read_text())
+    return custom_preprompts_path
+
+
+@app.command()
+def main(
+    project_path: str = typer.Argument("projects/example", help="path"),
+    model: str = typer.Argument("gpt-4-1106-preview", help="model id string"),
+    temperature: float = 0.1,
+    improve_mode: bool = typer.Option(
+        False,
+        "--improve",
+        "-i",
+        help="Improve code from existing project.",
+    ),
+    vector_improve_mode: bool = typer.Option(
+        False,
+        "--vector-improve",
+        "-vi",
+        help="Improve code from existing project using vector store.",
+    ),
+    lite_mode: bool = typer.Option(
```

Comment:
From a user's perspective, what's the difference between "improving code" and "improving code using a vector store"?

I would expect the user intent to be "improve my code" and using a vector db or not is a technical implementation detail that should not be exposed to the user?

Reply:
As long as its not explained to the user what the difference is, I agree 100%. 

Concretely, improve requires the user to provide which files to change, whereas vector improve improve picks code chunks to improve automatically from the whole code base. Probably for a UX perspective, maybe the top-level names for should be something like improve_manual_file_selection and improve_automatic_file_selection? Quite bulky... What is your opinion @TheoMcCabe  ?

Additional info here is that vector improve should be seen more as a first implementation and is not a mature default workflow.

Reply:
Or `improve_all` and `improve_files`